### DISCUSSION TITLE
Is there way to filter by aggregate

### DISCUSSION THREAD

**Discussion by mithatercan** [prisma/prisma#16432] [Q&A] [answered]
I have a question about grouping guests' actions by their id and the total amount of their orders. This query returns the sum of guests order amount in `_sum.orderAmount`. However, I can't filter by `_sum.orderAmount`. Is it possible to filter in this manner?

```javascript
  const actions = await prisma.action.groupBy({
    where: {
      type: 'EARN',
    },
    by: ['guestId'],
    _sum: {
      orderAmount: true,
    },
    orderBy: {
      _sum: {
        orderAmount: 'desc',
      },
    },
  })
```
I'd like to do something similar to this but am unable to.

```javascript
 const actions = await prisma.action.groupBy({
    where: {
      type: 'EARN',
      _sum: {
        orderAmount: {
          gt: 100,
        },
      },
    },
    by: ['guestId'],
    _sum: {
      orderAmount: true,
    },
    orderBy: {
      _sum: {
        orderAmount: 'desc',
      },
    },
  })
```

**Comment by pantharshit00** [comment, marked as answer]
Hello @mithatercan,

I think you can filter by `_sum.orderAmount` using the `having` key: https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#filter-groups-with-having. Let me know if this works for you. If you still unable to produce desired results, you can share your schema and I can try to form a query.

**Reply by mithatercan** [reply]
Hello, and thank you for taking the time to reply to my question.

I've already tried using the having key, but it returns the following error: `Unknown arg _sum in having._sum`. As far as I understand, prisma does not support filtering by aggregations.

**Reply by mithatercan** [reply]
@pantharshit00 Oh my bad, `having` key solve my problem, thank you again! 

```javascript

const actions = await prisma.action.groupBy({
    _sum: {
      orderAmount: true,
    },
    by: ['guestId'],
    where: {
      type: 'EARN',
    },
    having: {
      orderAmount: {
        _sum: {
          gt: 340.0,
        },
      },
    },
  })

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
